### PR TITLE
Cleaning up MA APIs before GA

### PR DESCRIPTION
**Pull request by anuchandy**
- Removal of the word "Anomaly" from (get|list|create|deelte|update)AnomalyAlertConfig() APIs.
- Rename of MetricAnomalyAlertConfiguration to MetricAlertConfiguration.
- Rename DetectionConditionsOperator to DetectionConditionOperator
- Rename AnomalyAlertConfiguration.dimensionsToSplitAlert property to dimensionsToSplitAlert
- Adding [get|set]adminEmail() to Hook
- Removal of add** methods (that adds to list properties) from couple of models

### CODE REVIEW

File: sdk/metricsadvisor/azure-ai-metricsadvisor/README.md [hunk 1]
```diff
@@ -320,29 +317,29 @@ final AnomalyAlertConfiguration anomalyAlertConfiguration
...
-            alertConfigurationId,
-            alert.getId())
-            .forEach(anomaly -> {
-                System.out.printf("DataPoint Anomaly was created on: %s%n", anomaly.getCreatedTime());
-                System.out.printf("DataPoint Anomaly severity: %s%n", anomaly.getSeverity().toString());
-                System.out.printf("DataPoint Anomaly status: %s%n", anomaly.getStatus());
-                System.out.printf("DataPoint Anomaly related series key: %s%n", anomaly.getSeriesKey().asMap());
-            });
-    });
+ng alertConfigurationId = "9ol48er30-6e6e-4391-b78f-b00dfee1e6f5";
```
Comment (samvaity) [inline sdk/metricsadvisor/azure-ai-metricsadvisor/README.md:322]: This seems to been cut off weirdly.

File: sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/implementation/models/AzureSQLConnectionStringParam.java [hunk 2]
```diff
@@ -13,7 +13,7 @@
     /*
      * The connection string to access the Azure SQL.
      */
-    @JsonProperty(value = "connectionString", required = true)
+    @JsonProperty(value = "connectionString")
```
Comment (samvaity) [inline sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/implementation/models/AzureSQLConnectionStringParam.java:16]: Is this a swagger update?
Does this change the way SDK's require params when creating a DataFeed?

File: sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/implementation/models/HookInfo.java [hunk 3]
```diff
@@ -132,4 +132,15 @@ public HookInfo setExternalLink(String externalLink) {
...
         return this.admins;
     }
+
+    /**
+     * Set the admins property: hook administrators.
+     *
+     * @param admins the admins value to set.
+     * @return the HookInfo object itself.
+     */
+    public HookInfo setAdmins(List<String> admins) {
```
Comment (samvaity) [inline sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/implementation/models/HookInfo.java:142]: Should this be an unmodifiable list?

File: sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/implementation/models/HookInfo.java [hunk 4]
```diff
@@ -132,4 +132,15 @@ public HookInfo setExternalLink(String externalLink) {
     public List<String> getAdmins() {
         return this.admins;
     }
+
+    /**
+     * Set the admins property: hook administrators.
```
Comment (samvaity) [inline sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/implementation/models/HookInfo.java:137]: Need to update the javadoc to make it more clear.

File: sdk/metricsadvisor/azure-ai-metricsadvisor/src/samples/java/com/azure/ai/metricsadvisor/administration/MetricsAnomalyAlertConfigOperationsAsyncSample.java [hunk 5]
```diff
@@ -86,9 +88,11 @@ public static void main(String[] args) {
         // Update the anomaly alert config.
         Mono<AnomalyAlertConfiguration> updatedAlertConfigMono = fetchAnomalyAlertConfig
             .flatMap(anomalyAlertConfig -> {
-                return advisorAdministrationAsyncClient.updateAnomalyAlertConfig(
+                List<String> hookIds = new ArrayList<>(anomalyAlertConfig.getIdOfHooksToAlert());
+                hookIds.remove(hookId2);
+                return advisorAdministrationAsyncClient.updateAlertConfig(
                     anomalyAlertConfig
-                        .removeHookToAlert(hookId2)
+                        .setIdOfHooksToAlert(hookIds)
```
Comment (samvaity) [inline sdk/metricsadvisor/azure-ai-metricsadvisor/src/samples/java/com/azure/ai/metricsadvisor/administration/MetricsAnomalyAlertConfigOperationsAsyncSample.java:95]: nit: rename suggestion setHookIdsToAlert ?

Comment (anuchandy) [reply sdk/metricsadvisor/azure-ai-metricsadvisor/src/samples/java/com/azure/ai/metricsadvisor/administration/MetricsAnomalyAlertConfigOperationsAsyncSample.java:95]: thanks, yes this sounds better, will update.

File: sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/implementation/models/AzureSQLConnectionStringParam.java (hunk 2 above)
Comment (anuchandy) [reply sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/implementation/models/AzureSQLConnectionStringParam.java:16]: Yes, it's a swagger update. The reason for the service team changing it is, it's an output model as well but in response `connectionString` will be absent. But of course, when creating it is required.

File: sdk/metricsadvisor/azure-ai-metricsadvisor/README.md (hunk 1 above)
Comment (anuchandy) [reply sdk/metricsadvisor/azure-ai-metricsadvisor/README.md:322]: ah.. something went wrong when I ran npx, checking

File: sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/implementation/models/HookInfo.java (hunk 3 above)
Comment (anuchandy) [reply sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/implementation/models/HookInfo.java:142]: This is input list, only in getter we apply `Collections.unmodifiableList(..)`

File: sdk/metricsadvisor/azure-ai-metricsadvisor/README.md [hunk 6]
```diff
@@ -320,20 +317,20 @@ final AnomalyAlertConfiguration anomalyAlertConfiguration
 ```
 #### Query anomaly detection results
 This example demonstrates how a user can query alerts triggered for an anomaly detection configuration and get anomalies for that anomalyAlert.
-
+<!-- embedme ./src/samples/java/com/azure/ai/metricsadvisor/ReadmeSamples.java#L256-L276 -->
 ```java
-String alertConfigurationId = "9ol48er30-6e6e-4391-b78f-b00dfee1e6f5";
+long alertConfigurationId = "9ol48er30-6e6e-4391-b78f-b00dfee1e6f5";
```
Comment (samvaity) [inline sdk/metricsadvisor/azure-ai-metricsadvisor/README.md:322]: Should the value type here still be String?

File: sdk/metricsadvisor/azure-ai-metricsadvisor/README.md [hunk 7]
```diff
@@ -320,20 +317,20 @@ final AnomalyAlertConfiguration anomalyAlertConfiguration
...
 final OffsetDateTime startTime = OffsetDateTime.parse("2020-01-01T00:00:00Z");
 final OffsetDateTime endTime = OffsetDateTime.parse("2020-09-09T00:00:00Z");
-metricsAdvisorClient.listAlerts(
+MetricsAdvisorClient.listAlerts(
```
Comment (samvaity) [inline sdk/metricsadvisor/azure-ai-metricsadvisor/README.md:325]: This change should be reverted?

File: sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/implementation/models/HookInfo.java (hunk 4 above)
Comment (anuchandy) [reply sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/implementation/models/HookInfo.java:137]: This is an auto-gen file, but we have a better documentation in the hand-written type wrapping this file (NotificationHook) like below:

> The list of user e-mails with administrative rights to manage the hook.

File: sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/administration/models/ListDetectionConfigsOptions.java [hunk 8]
```diff
@@ -51,7 +51,7 @@ public ListMetricAnomalyDetectionConfigsOptions setMaxPageSize(int maxPageSize)
      * @param skip The skip value.
      * @return ListMetricAnomalyDetectionConfigsOptions itself.
```
Comment (samvaity) [inline sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/administration/models/ListDetectionConfigsOptions.java:52]: Need to change the return type name in docs

File: sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/administration/models/ListDataFeedIngestionOptions.java [hunk 9]
```diff
@@ -72,7 +72,7 @@ public Integer getSkip() {
      * @param maxPageSize The skip value.
```
Comment (samvaity) [inline sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/administration/models/ListDataFeedIngestionOptions.java:72]: incorrect doc here, probably a change before this PR.

File: sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/implementation/models/HookInfo.java (hunk 4 above)
Comment (samvaity) [reply sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/implementation/models/HookInfo.java:137]: Oh its implementation file, please ignore.

**Review by samvaity — APPROVED** [review]

File: sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/administration/models/ListDataFeedIngestionOptions.java (hunk 9 above)
Comment (anuchandy) [reply sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/administration/models/ListDataFeedIngestionOptions.java]: thanks, yep, I just went through all files and ensured we have consistent doc for maxPageSize

File: sdk/metricsadvisor/azure-ai-metricsadvisor/README.md (hunk 6 above)
Comment (anuchandy) [reply sdk/metricsadvisor/azure-ai-metricsadvisor/README.md]: thanks, corrected

File: sdk/metricsadvisor/azure-ai-metricsadvisor/README.md (hunk 7 above)
Comment (anuchandy) [reply sdk/metricsadvisor/azure-ai-metricsadvisor/README.md]: thanks, corrected

**Pull request merged**